modules/core/src/convert.cpp

```diff
@@ -4643,24 +4643,35 @@ cvtScaleHalf_<short, float>( const short* src, size_t sstep, float* dst, size_t
 #ifdef HAVE_OPENVX
 template<typename T, typename DT>
 static bool _openvx_cvt(const T* src, size_t sstep,
-                        DT* dst, size_t dstep, Size size)
+                        DT* dst, size_t dstep, Size continuousSize)
 {
     using namespace ivx;
 
-    if(!(size.width > 0 && size.height > 0))
+    if(!(continuousSize.width > 0 && continuousSize.height > 0 && sstep > 0 && dstep > 0))
     {
         return true;
     }
 
+    //.height is for number of continuous pieces
+    //.width  is for length of one piece
+
+    Size imgSize = continuousSize;
+    if(continuousSize.height == 1)
+    {
+        //continuous case
+        imgSize.width  = sstep / sizeof(T);
```

Comment:
are you sure that `sstep / sizeof(T) == dstep / sizeof(DT)` ?
don't you think `assert` would be sensible?